android/src/main/java/io/ably/flutter/plugin/AblyMessageCodec.java

```diff
@@ -110,4 +207,44 @@ private Object readTokenDetails(final ByteBuffer buffer) {
 
         return o;
     }
+
+
+    //HANDLING WRITE
+    private void writeErrorInfo(ByteArrayOutputStream stream, ErrorInfo e){
+        stream.write(_errorInfo);
+        writeValue(stream, e.code);
+        writeValue(stream, e.message);
+        writeValue(stream, e.statusCode);
+        writeValue(stream, e.href);
+        writeValue(stream, null); //requestId - not available in ably-java
+        writeValue(stream, null); //cause - not available in ably-java
+    }
+
+    private void writeConnectionEvent(ByteArrayOutputStream stream, ConnectionEvent e){ writeEnum(stream, _connectionEvent, e); }
+    private void writeConnectionState(ByteArrayOutputStream stream, ConnectionState e){ writeEnum(stream, _connectionState, e); }
+    private void writeConnectionStateChange(ByteArrayOutputStream stream, ConnectionStateListener.ConnectionStateChange c){
+        stream.write(_connectionStateChange);
+        writeValue(stream, c.current);
+        writeValue(stream, c.previous);
+        writeValue(stream, c.event);
+        writeValue(stream, c.retryIn);
+        writeValue(stream, c.reason);
+    }
+
+    private void writeChannelEvent(ByteArrayOutputStream stream, ChannelEvent e){ writeEnum(stream, _channelEvent, e); }
+    private void writeChannelState(ByteArrayOutputStream stream, ChannelState e){ writeEnum(stream, _channelState, e); }
+    private void writeChannelStateChange(ByteArrayOutputStream stream, ChannelStateListener.ChannelStateChange c){
+        stream.write(_channelStateChange);
+        writeValue(stream, c.current);
+        writeValue(stream, c.previous);
+        writeValue(stream, c.event);
+        writeValue(stream, c.resumed);
+        writeValue(stream, c.reason);
```

Comment:
As mentioned before in my other PRs, we are now making this problem worse, keeping the ordering of these stream writes and reads in sync.  I really dislike this.